### PR TITLE
Integrate Nimbora API for accurate APR calculation

### LINKED ISSUE #281: Add base APRs for Nimbora
There is the following problem in `src/endpoints/discover/defi/get_alt_protocol_stats.rs`: we only count DeFi Spring APRs. This route should return the APRs of different strategies, but it is currently only using the APRs provided by the DeFi Spring for these strategies. The aim of this issue is to add an API call to Nimbora at the begining of the file, so that the APR of every strategy returned by the DeFi Spring API is replaced by the one from the Nimbora API (which counts the base APR + the DeFi Spring one).

Example to illustrate:
When  calling`https://api.starknet.quest/discover/defi/get_alt_protocol_stats` (which is the current state of the API), we get:
image

And when calling the Nimbora API `https://stats.nimbora.io/yield-dex/strategies`:
image 
We get, (this screenshot only contains this example strategy, but all other strategies are returned by the API), for instance,
``` "aprBreakdown": {
      "base": "0.7026797371850104",
      "boost": "0",
      "incentives": "13.09410701089488"
    },
    "apr": "13.7967867480798904",
```
The problem is that our API only count the `incentives`(returned by the DeFi Spring API), and not the base. That's why you'll need to use both API's, to get the accurate computer APR.

You'll need to set the following field in you `config.toml` in order to test:
```toml
alt_protocols_api_endpoint = "https://kx58j6x5me.execute-api.us-east-1.amazonaws.com/starknet/fetchFile?file=prod-api/phase4/phase4_strk_grant.json"
```

Suggested implementation:
- At the begining of the route, call the Nimbora's API to fetch strategies APRs.
- At the end of the route, just before returning the response, map every strategy, and find (if available => because in the future we could have other protocols than Nimbora, so the Nibmora's API wouldn't have the data for them), in the stored response from Nimbora, the related strategy. Then collect and update the APR.
- If there is any issue when calling Nimbora's API, just ignore its response and proceed without this data (meaning you only use the one from DeFi spring as we are currently doing)

### PR DESCRIPTION
**Pull request by Benjtalkshow**
Closes #281 
### Key Updates:
1. Introduced an async function `fetch_nimbora_data()` to pull data from the Nimbora API.
2. Updated the handler in `get_alt_protocol_stats.rs` to:
   - Fetch Nimbora data at the start.
   - Process DeFi Spring data as usual.
   - Before returning, update matching strategies with both base APR and incentive APR from Nimbora.
3. Added error handling to ensure the system continues using just DeFi Spring data if the Nimbora API fails.
4. Included basic logging to monitor strategy matching between DeFi Spring and Nimbora.

### CODE REVIEW

File: src/endpoints/discover/defi/get_alt_protocol_stats.rs [hunk 1]
```diff
@@ -10,42 +10,87 @@ use serde_json::Value;
...
+                    }
+                }
 
-                                let len = &arr_value.unwrap().len();
-                                if len == &0 {
-                                    continue;
+                // Update APRs with Nimbora data if available
+                if let Ok(Value::Array(nimbora_strategies)) = nimbora_data {
+                    for (protocol, strategies) in new_map.iter_mut() {
+                        println!("Processing protocol: {}", protocol);
```
Comment (Marchand-Nicolas) [inline src/endpoints/discover/defi/get_alt_protocol_stats.rs]: Please remove prints when there is no error (you can keep the error printing)

File: src/endpoints/discover/defi/get_alt_protocol_stats.rs [hunk 2]
```diff
@@ -10,42 +10,87 @@ use serde_json::Value;
...
+                                if let Some(nimbora_strategy) = nimbora_strategies.iter().find(|s| {
+                                    s["symbol"].as_str().unwrap_or("") == nimbora_symbol
+                                }) {
+                                    if let Some(apr) = nimbora_strategy["apr"].as_str() {
+                                        if let Ok(apr_value) = apr.parse::<f64>() {
+                                            strategy_data["apr"] = Value::Number(
+                                                serde_json::Number::from_f64(apr_value)
+                                                    .unwrap_or(serde_json::Number::from(0)),
+                                            );
+                                            println!("  Updated APR for {} ({}) to {}", strategy_name, nimbora_symbol, apr);
```
Comment (Marchand-Nicolas) [inline src/endpoints/discover/defi/get_alt_protocol_stats.rs]: Same here

**Review by Marchand-Nicolas — CHANGES_REQUESTED** [review]
Working perfectly well done! Just remove some println

**Review by Marchand-Nicolas — CHANGES_REQUESTED** [review]
Working perfectly well done! Just remove some println

**Comment by Benjtalkshow** [comment]
Hi @Marchand-Nicolas 
I have made the requested changes

**Review by Marchand-Nicolas — APPROVED** [review]
Lgtm

**Pull request merged**